_episodes/09-cmorization.md

```diff
@@ -0,0 +1,1053 @@
+> ```
+>
+> For that to happen, you will have to fix/work on the following things:
+> - adding standard names to the dimensions ``lat`` and ``lon``
+> - fix the metadata for the variable "gpp"
+> - change the time units to start in the year 1950
+> - make the coordinates CMOR-compliant
+> - set global attributes for the data file
+>
+> > ## Answers
+> >
+> > To fully CMORize the "FLUXCOM" dataset, you need to add the following
+> > lines of code to the ``_extract_variable`` function of your CMORizer
+> > script:
+> >
+> > ```python
+> > def _extract_variable(cmor_info, attrs, filepath, out_dir):
+> >     """Extract variable."""
+> >     var = cmor_info.short_name
+> >     logger.info("Var is %s", var)
+> >     cubes = iris.load(filepath)
+> >     for cube in cubes:
+> >         # convert data from gc/m2/day to kg/m2/s
+> >         cube = cube / (1000 * 86400)
+> >         cube.units = 'kg m-2 s-1'
+> >
+> >         # The following two lines are needed for iris.util.guess_coord_axis
+> >         cube.coord('lat').standard_name = 'latitude'
+> >         cube.coord('lon').standard_name = 'longitude'
+> >         utils.fix_var_metadata(cube, cmor_info)
+> >         utils.convert_timeunits(cube, 1950)
+> >         utils.fix_coords(cube)
+> >         utils.set_global_atts(cube, attrs)
+> >         utils.flip_dim_coord(cube, 'latitude')
+> >         coord = cube.coord('latitude')
+> >         coord.bounds = np.flip(coord.bounds, axis=1)
+> >         logger.info("Saving file")
+> >         utils.save_variable(cube,
+> >                             var,
+> >                             out_dir,
+> >                             attrs,
+> >                             unlimited_dimensions=['time'])
+> >
+> > ```
+> >
+> {: .solution}
+{: .challenge}
+
+So now we can finally run our new CMORizing script to produce an ESMValTool
+readable datafile of the "FLUXCOM" dataset. As already mentioned, the call
+to run a CMORizing script is as follows:
+
+```bash
+cmorize_obs -c <config-user.yml> -o <dataset-name>
+```
+
+If your run was successful, you should have gotten no error message in the
+ESMValTool logging information and one or more NetCDF files should have been
+produced in your output directory.
+
+To make sure that the CMORization has really worked as planned, we run the
+CMOR checker on this newly produced NetCDF file. If the ESMValTool can read
+the data without problems, you should see as one of the last lines of
+ESMValTool output:
+
+```bash
+INFO    Run was successful
+```
+
+## Last steps
+
+Congratulations! You have successfully CMORized a new dataset!
+Since you have gone through all the trouble to reformat the dataset so that
+the ESMValTool can work with it, it would be great if you could provide the
+CMORizer, and ultimately with that the dataset, to the rest of the community.
+To do that there are a few more steps you have to do:
+1. Open a pull request in the ESMValTool repository describing the dataset briefly
```

Comment:
We encourage users to discuss their ideas in an issue as the first step of contribution. So, I suggest replacing this step with a sentence pointing to the episode 08-development-setup where the contribution procedure is explained.

Reply:
Done